### PR TITLE
Improved jam avoidance by improving MovableStrategy.findWayAroundObstacle()

### PR DESCRIPTION
**Pull request by andreas-eberle**

### CODE REVIEW

**Comment by andreas-eberle** [comment]
@codingberlin: In my commit, I improved the calculations how movables find ways around other movables that cannot be pushed away (e.g. Bricklayers when they are construction a building). This greatly reduces visible jams. 

However, two AI tests fail now. I had a look at them and it looks like these changes made the AI a lot weaker. But I have no idea how I could have caused that and it looks good when I let the AI fight each other on regular maps. 

When I compared the behaviour of the AI in the "hardShouldConquerEasy" test, I saw that the stronger AI rushes the weaker one with the code of the current master. With the code of this branch, the stronger AI does not rush the weaker one resulting in a 3 hours fight between them with very even strengths... 

What do you think? Can you have a look?

**Comment by codingberlin** [comment]
@andreas-eberle I'll look into this.

My first thought:
This shows that every small improvement or change of the game changes the balancing.
The jams before the fix seemed to influence the easy ai to be a bit less stronger so that the hard ai decided to attack and thus win the game very early.
The red test (or after 3 hours green test) shows us, that the easy and hard ai are very close in its strengths (Both have the same economy manager, but different army general)
I suggest to ether @Ignore the test or better update it to 3 hours and create an issue "update balance of AI's so that their difficulties are really different levels".

**Comment by homoroselaps** [comment]
Hey I based my pull request on this pull request and was able to fix both parts of #309. The tests still fail because of the same reasons as before.
Therefore I suggest closing this pull request in favor of mine.

**Comment by andreas-eberle** [comment]
@homoroselaps: I plan on merging this PR into master, after I made some final tests. I will then look at your PR to add those changes later. Its better to have smaller PRs.

File: jsettlers.tests/tests/jsettlers/tests/ai/AiDifficultiesIT.java [hunk 1]
```diff
@@ -64,7 +65,7 @@ public void veryHardShouldConquerHard() {
 	}
 
 	@Test
-	public void verHardShouldProduceCertainAmountOfSoldiersWithin75Minutes() {
+	public void verHardShouldProduceCertainAmountOfSoldiersWithin85Minutes() {
```
Comment (homoroselaps) [inline jsettlers.tests/tests/jsettlers/tests/ai/AiDifficultiesIT.java]: is this a typo? ver(y)HardShouldProduce

Comment (andreas-eberle) [inline jsettlers.tests/tests/jsettlers/tests/ai/AiDifficultiesIT.java]: yes it is, thanks.

**Comment by andreas-eberle** [comment]
@michaelzangl: Could you have a look at the changes, so I can merge it?

File: jsettlers.common/resources/buildings/big_livinghouse.xml [hunk 2]
```diff
@@ -118,30 +116,26 @@
 	<blocked dx="-5" dy="-3" block="false" />
 	<blocked dx="-5" dy="-2" block="false" />
 	<blocked dx="-5" dy="-1" block="false" />
-	<blocked dx="-4" dy="1" block="false" />
```
Comment (michaelzangl) [inline jsettlers.common/resources/buildings/big_livinghouse.xml]: Mind that removing those will lead to less space in between the houses.

**Comment by michaelzangl** [comment]
Looks fine to me

File: jsettlers.common/resources/buildings/big_livinghouse.xml (hunk 2 above)
Comment (andreas-eberle) [inline jsettlers.common/resources/buildings/big_livinghouse.xml]: I know, but I checked that there is still a closed protected area around the building.

Comment (michaelzangl) [inline jsettlers.common/resources/buildings/big_livinghouse.xml]: You will have less space for your bearers to move, especially when constructing the building. This is why there was that extra protected area. But we'll see how it works and if the jam issue is improved or not by this.

**Pull request merged**